### DISCUSSION TITLE
Password longer that 8 characters

### DISCUSSION THREAD

**Discussion by louigi600** [python-smpplib/python-smpplib#228] [Q&A]
I realise this might be in contrast to SMPP standard but I had a rea life scenario with SMSC that wants me to send a password that is longer that 8 characters, I know I have the correct password but I'm unable to bing with error "incorrect password". 
I'm assuming it is because the password gets truncated.
I was looking at the code to see if I could force it to be longer and found this:

smpplib/command.py:        'password': Param(type=str, max=9),

Maybe if I put that to something long enough for my needs it will be enough ?

How do I force reinstall after changing the code ?

will running

sudo rm -rf build
sudo python ./setup.py install
 
do the trick ?

**Comment by eigenein** [comment]
How long is your password?

As for the reinstallation, please refer to the [Installing Packages](https://packaging.python.org/en/latest/tutorials/installing-packages/#installing-from-a-local-src-tree)

**Comment by louigi600** [comment]
Well I think we have out of spec password length ... 12 if I remember
right. I modified the code as I suggested and reinstalled ... it worked.
There is however some issue with the script not always intercepting
properly the delivery.

**Comment by louigi600** [comment]
I gave made a few other modifications get it to quit properly:
rewrote the delivered handler:
```
def handle_deliver_sm(pdu):
        sys.stdout.write('delivered {}\n'.format(pdu.receipted_message_id))
        client.unbind()
        client.disconnect()
        sys.exit()
        return 0 # cmd status for deliver_sm_resp
```
        
 Also I do with smpplib in the main loop

with smpplib.client.Client(SMSC_HOST, SMSC_PORT) as client:


All the hard coded stuff is in variables now foe easier edit


BTW: although it it written somewhere that it should work with python 2.7 ... I was unable to get smpplib to work with python 2.7
